scripts/apps/monitoring/helpers/CustomSortOfGroups.ts

```diff
@@ -13,13 +24,38 @@ export function matchGroupToOrderConfig(group: StageGroup) {
     return 'monitoring.stage.sort';
 }
 
-export default function getCustomSortForGroup(config: any, group: StageGroup): GroupSortOptions {
+export default function getCustomSortForGroup(config: any, group: StageGroup): GroupSortConfig | null {
     if (!group || !group._id || !group.type) {
-        return [];
+        return null;
     }
 
     const configForGroup = matchGroupToOrderConfig(group);
-    const customConfig = get(config, configForGroup, []);
+    const customConfig: GroupSortConfig = get(config, configForGroup, null);
+
+    if (customConfig != null && customConfig.default) {
+        const {field, order} = getDefaultFieldForConfig(customConfig);
+
+        customConfig.default = `${field}:${order}`;
+    }
 
     return customConfig;
 }
+
+export function getDefaultFieldForConfig(config: GroupSortConfig):
+    {field: string, order: OrderType} | null {
+    let [field, order] = config.default.split(':');
+
+    if (!order || !isOrderType(order)) {
+        order = 'asc';
+    }
+
+    // make sure default field exists in options
+    if (!config.options.includes(field)) {
+        return null;
+    }
+
+    return {
+        field,
+        order: order as OrderType,
```

Comment:
Don't use `as`. Use `if/else` so the compiler can verify that you are passing a correct type.

```

export function getDefaultFieldForConfig(config: GroupSortConfig):
    {field: string, order: OrderType} | null {
    let [field, order] = config.default.split(':');

    // make sure default field exists in options
    if (!config.options.includes(field)) {
        return null;
    }

    if (!order || !isOrderType(order)) {
        order = 'asc';
    } else {
        return {
            field,
            order: order,
        };
    }
}

```

Reply:
Good catch. I assume after `order = 'asc'` you wanted to include the `return` too?

Reply:
yep, forgot a return.